velox/docs/develop/window.rst

```diff
@@ -205,3 +205,109 @@ peer row with a NULL and the frame_end index is the last peer row with a NULL va
 
 Rows with NULL values do not participate in the frames of the other rows.
 
+
+Empty frames
```

Comment:
nit: Maybe "Empty and partial frames" since we talk about partial frames in this section too?

Reply:
@kagamiori : So the thing is that the function author needs to understand the concept of empty frames and the corresponding validFrames function parameter in the API. 

Partial frames need to be called out. But since the framework handles them the function author as such doesn't need to understand or handle them. 

So I feel its better to call this section "Empty frames" and have a sub-section for partial frames.

Reply:
@aditi-pandit, then could you clarify at the beginning of this section that empty frames need to be handled explicitly by window function authors while partial frames do not? Also, could we put the subsection "Handling empty frames in window functions" together with the introduction of empty frames, and put the subsection "Partial frames" after that?

Reply:
@kagamiori : I did some rewording where I introduced the type of frames in the beginning stating that empty frames need special handling. Then the section goes on to describe the frame types and the final section has instructions on how to handle empty frames.

The Empty frames heading is inline with other headings in this doc Window frames, k Range frames and empty frames.

Reply:
@aditi-pandit, is it possible to move the "Partial frames" subsection after the "Handling empty frames in window functions" subsection? I think it would be clearer to talk about empty frames and handling them in window functions, then partial frames and that they don't need special handling in window functions. What do you think?

Reply:
@kagamiori : There is the diagram which visually shows all 3 so I felt its better to have all the definitions and then the handling section at the end. Else the diagram shows partial frames without any definition for it.

The last line of the Handling emtpy frames section also states that there is no special handling for partial frames again. So it is self-contained as well.

Reply:
OK, this makes sense.